src/software/simulated_tests/simulated_play_test_fixture.cpp

```diff
@@ -42,9 +43,23 @@ void SimulatedPlayTestFixture::setAIPlay(const std::string& ai_play)
     ai_control_config->getMutableCurrentAiPlay()->setValue(ai_play);
 }
 
+void SimulatedPlayTestFixture::setRefereeCommand(
+    const RefereeCommand& current_referee_command,
+    const RefereeCommand& previous_referee_command)
+{
+    game_state.updateRefereeCommand(previous_referee_command);
+    game_state.updateRefereeCommand(current_referee_command);
+}
+
+void SimulatedPlayTestFixture::setGameState(const GameState& game_state_)
+{
+    game_state = game_state_;
+}
+
 void SimulatedPlayTestFixture::updatePrimitives(
-    const World& world, std::shared_ptr<Simulator> simulator_to_update)
+    World world, std::shared_ptr<Simulator> simulator_to_update)
```

Comment:
correct me if I'm wrong but in the presentation a few weeks ago, wasn't passing world (and creating a copy) something we want to avoid?

Reply:
that's true, I needed a copy to update the gamestate, so I made the copy in the local context instead